scripts/plot_operator_breakdown.py

```diff
@@ -102,36 +176,15 @@
         "",
         "/\\/\\/\\/\\/\\",
     )
-    hatches = [p for p in patterns for i in range(len(df))]
-    for bar, hatch in zip(ax.patches, hatches):
-        # Calculate color so that the hatches are visible but not pushy
-        hsv = mplcolors.rgb_to_hsv(bar.get_facecolor()[:3])
-        hatch_color_hsv = hsv
-        hatch_color_hsv[2] = hsv[2] + 0.2 if hsv[2] < 0.5 else hsv[2] - 0.2
-        bar.set_edgecolor(mplcolors.hsv_to_rgb(hatch_color_hsv))
-
-        bar.set_hatch(hatch)
-        bar.set_linewidth(0)
-
-# Set labels
-ax.yaxis.set_major_formatter(ticker.PercentFormatter(xmax=1.0))
-ax.set_ylabel("Share of run time\n(Hiding ops <1%)")
-
-# Reverse legend so that it matches the stacked bars
-handles, labels = ax.get_legend_handles_labels()
-
-if paper_mode:
-    # Plot the legend under the graph (good for papers)
-    box = ax.get_position()
-    ax.set_position([box.x0, box.y0, box.width, box.height * 0.8])
-    legend = ax.legend(reversed(handles), reversed(labels), loc="lower center", ncol=3, bbox_to_anchor=(0.5, -0.45))
-
-    print("Plotting in 'paper' mode (legend below graph, hatching) - remove 'paper' argument to change")
-else:
-    # Plot the legend to the right of the graph (good for screens)
-    legend = ax.legend(reversed(handles), reversed(labels), loc="center left", ncol=1, bbox_to_anchor=(1.0, 0.5))
-    print("Plotting in 'screen' mode (legend on right, no hatching) - use 'paper' argument to change")
-
-# Layout and save
-plt.tight_layout()
-plt.savefig("operator_breakdown.pdf", bbox_extra_artists=(legend,), bbox_inches="tight")
+    for axis in [ax_1, ax_2, ax_3, ax_4]:
+        column_count = len(axis.get_xticks())
+        hatches = [p for p in patterns for i in range(column_count)]
+        for bar, hatch in zip(reversed(axis.patches), hatches):
+            bar.set_edgecolor(get_hatch_color(bar.get_facecolor()[:3]))
+            bar.set_hatch(hatch)
+            bar.set_linewidth(0)
+
+handles, labels = ax_1.get_legend_handles_labels()
+fig.legend(reversed(handles), reversed(labels), loc=9, ncol=10)
```

Comment:
Seeing it in action, I would rather place the legend at the bottom. What do you think?

Reply:
Personally, I don't like to see multiple plots without a legend on top. It's different if it is only a single plot. I can live with both positions.

Reply:
Won't argue.